rfcs/0098-community-team.md

```diff
@@ -0,0 +1,215 @@
+
+## Contacting the community team
+[contact]: #contact
+
+The community team will be contactable via a public email address listed on the NixOS website. This email address will forward to all community team members. Additionally, all community team members will list some form of individual contact information so community members can reach out to them directly in the case of an issue concerning other community team members or matters they are not comfortable disclosing to the entire community team.
+
+Emails sent to the community team are not to be shared outside of the community team, or to more people than need to be involved in the case of messages sent to individual team members.
+
+There will be a method to quickly notify all community team members in official public communication channels, in case of something needing urgent attention.
+
+Any of these contact methods may be used for disputes, for discussion of issues facing the community or individual members, as a way to raise safety concerns, and for any other reason that feels necessary.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This effort will be a focus of controversy, especially at the outset. It is the position of these authors that it's a necessary controversy, and that things will get less intense once they settle.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Failing to actively manage and lead the community makes it difficult for the community to practice any sort of values, not even seemingly "easy" values such as inclusion of minorities and an opposition to fascism. Every technical project is, necessarily, also a community, and it is important to live up to that responsibility for the betterment of society as a whole.
+
+Creating a hierarchical power structure based on specific individuals having semi-permanent roles is an approach that could initially be successful at reducing conflict. However, it would require a great deal of trust which may not exist, and it is not sustainable to indefinitely place that level of trust in specific people. By having a rotating team which includes *everyone*, we spread the necessary skills throughout the community and foster a culture which handles conflict in appropriate ways from the beginning.
+
+A suggestion from @blaggacao:
+* Submit one procedural RFC for the Moderating(!) Team
+* A normative RFC for our "Statement of Values" which can be discussed without this really incommodating defensive framing of "policing discussions on the public forum"
+* As prominent member pointed out, the NixOS foundation already had had a tweet commiting to the entirety of prerequisite principles of emotional safety in public discussions. An really short RFC could take that statement, maybe expand a little bit in context and move it out of twitter into the body of (accepted) RFCs. Such RFC will have "teeth".
+
+This suggestion is included verbatim, per discussion on the GitHub thread.
+
```

Comment:
Another alternative could be to use RFC #114 as a replacement for the goals and values sections (most widely disputed part of the RFC); and transition this RFC into a more focused discussions around the creation of a moderation body.

Reply:
I don't think it's a good idea to split up the values and community team aspects, out of two primary concerns:
- With there being considerable disagreement and/or misunderstanding over the shape of the community team, it's likely that this RFC would remain in limbo for longer than a CoC/values RFC would. That'd lead to a situation where there is a defined CoC but no body to enforce it - which in turn is likely to do lasting damage to the reputation of the NixOS community because it does not live up to its promises from the perspective of an outsider (see eg. the current issues in the Scala community for an example).
- In many ways, the proposal for the community team (or at least the *intent* behind it, modulo how it is communicated) is *an implementation of* the values that are specified in this RFC. With both sections still in flux and subject to change based on community feedback, trying to pass the two separately is likely to lead to the proposals going 'out of sync' on fundamental values - which would be a significant problem for the proposed community team structure, as there is no hierarchy to correct for that after the fact. A non-hierarchical system like this can only work if there is a clear *and accurate* set of values according to which it operates.

Reply:
A separate values and moderation team RFCs would make it easier to keep the conversation focused. The #114 discussion is proceeding fairly well and has a chance for broad consensus.

Regarding the potential lasting damage to the reputation of the NixOS community, we should at least acknowledge that this RFC (and the ensuing discussion, bans, divisiveness, and rants) has already done some damage, and will likely do more. I'm not convinced that that damage has been worth it.

Regarding the implementation of a community team, we keep hearing it referenced as a non-hierarchical team structure; what does this label mean? How can this team reflect the community if it is self-selecting with no oversight? Placing the selection of the initial team on a single individual also seems problematic when the current moderation efforts are performed by more than that single individual. There is a very real risk that the community team itself will be out of sync with the needs of community. Codified values don't necessarily prevent a drift between the stated and de facto values.

Reply:
> The #114 discussion is proceeding fairly well and has a chance for broad consensus.

That doesn't automatically make it a good proposal, however. Covering less ground *necessarily* means it's easier to gain consensus, but if it doesn't actually address the problems that we're trying to solve, that isn't helpful. The priority should be to build a durable community team structure, not to go for the easiest consensus. That is necessarily going to involve some difficult conversations.

> Regarding the potential lasting damage to the reputation of the NixOS community, we should at least acknowledge that this RFC (and the ensuing discussion, bans, divisiveness, and rants) has already done some damage, and will likely do more. I'm not convinced that that damage has been worth it.

Yes, it has, but so far only internally. More importantly, that damage is not undone by dropping this proposal.

This reminds me of the discussion around "firing developers who break production". It's something people commonly call for because it seems a superficially obvious solution, but in reality it's the *worst* thing you could possibly do as a company - that developer has just learned a very expensive lesson, why would you fire them and not at least benefit from the education that came with the expenses?

I feel similarly about this situation. This RFC has surfaced a number of disagreements and conflicts, and through miscommunication has led to people holding certain beliefs about the intentions of others in the community. This sucks, but those beliefs are not going to go away now unless the work is done to mediate and bridge communication between people. Since that work now needs to be done *anyway*, why not do it in the context of this very RFC?

> Regarding the implementation of a community team, we keep hearing it referenced as a non-hierarchical team structure; what does this label mean? How can this team reflect the community if it is self-selecting with no oversight? Placing the selection of the initial team on a single individual also seems problematic when the current moderation efforts are performed by more than that single individual. There is a very real risk that the community team itself will be out of sync with the needs of community. Codified values don't necessarily prevent a drift between the stated and de facto values.

These are all legitimate concerns, but they are also concerns that the authors are willing to address (ie. a reason for updating *this* RFC, not just creating a competing one) - and, especially in the case of "what does non-hierarchical mean", the authors *are* already addressing it. It's my understanding that there is currently a significant rewrite of this RFC underway which among other things seeks to clarify the 'non-hierarchical' part, as it turns out that it's currently unclear to too many people.

(I thought that the in-progress clarification of "non-hierarchical" was already mentioned earlier in this RFC thread, but it seems that that is not the case. This *was* mentioned on Matrix, but it doesn't seem to have made it into this thread. Sorry for that.)

Reply:
> Yes, it has, but so far only internally. More importantly, that damage is not undone by dropping this proposal.

We never know if there is further damage to have from it (I do hope things improve after rewrite, but no idea if it will be enough), and conversion of damage to external is a higher risk for continuously  active discussions than for buried and closed ones.

> Covering less ground necessarily means it's easier to gain consensus, but if it doesn't actually address the problems that we're trying to solve, that isn't helpful.

> Since that work now needs to be done anyway, why not do it in the context of this very RFC?

Well, we have already converted from obvious inference to outright common knowledge that there is N>1 of subgroups that find the _beliefs_ of each other regrettable. There are _different_ directions of work that could be done in response. We can continue discussing values or we can restrict the scope to actions and behaviours. The current version of RFC _as a text_ is ambiguous on this choice; I think the former choice will lead to ongoing damage and for the latter #114 has a headstart if we compare the texts available today.

Reply:
I don't begrudge anyone promoting a preferred approach (say, #114), but do hope we can keep this RFC open too. I liked the original text for this RFC, and also look forward to reading the updated text.

With regards to RFC #114: Reading the conversation in #114 I noticed that many of the voices speaking up in favour of #98 did not (yet) share their ideas about #114. If consensus in RFC #114 seems closer, I think it might be because far fewer folks have spoken there. If everyone who contributed here joined the conversation in #114 too, I imagine the conversations in both RFCs would start looking very similar.

Reply:
One thing that hit me recently is that both the CoC and the current version of RFC don't reflect our *actual community values*. Even if we don't have anything written down right now, it's possible to observe emergent values behind exhibited throughout our community. Extracting and observing those would allow for a much smoother transition (and easier consensus) if we want the community team to "live and breathe" by our community values.

What I hear is that the current community is fine overall, and the concern is that future members would change the nature of our project. A good way to do that is to look at how we do things and make those first-class so they can be more easily communicated to new members.

When this hit me, I started this thread on Discourse: https://discourse.nixos.org/t/what-are-the-values-that-you-see-being-expressed-in-the-nixos-project/15959 . We all have a different perception of the community and it's interesting to find out what other people think.

Reply:
To go back to this thread, one practice/value we seem to be consistently following is trying to find subsets where we can generate strong consensus, and adopt those. Then iterate. Very large RFCs are fine to generate discussions, or to paint a picture, but it's perfectly legit to extract parts that we already agree on and merge those in as well. It doesn't have to be a one *or* the other discussion. If the end result is not internally consistent, we can even have an RFC replace another. This process is more democratic than asking the community to accept a large change wholesale.